jep/0000/README.adoc

```diff
@@ -0,0 +1,124 @@
+//
+//
+// Uncomment if discussion will occur in forum other than jenkinsci-dev@ mailing list.
+//| Discussions-To
+//| Link to where discussion and final status announcement will occur
+//
+//
+// Uncomment if this JEP depends on one or more other JEPs.
+//| Requires
+//| JEP-NUMBER, JEP-NUMBER...
+//
+//
+// Uncomment and fill if this JEP is rendered obsolete by a later JEP
+//| Superseded-By
+//| JEP-NUMBER
+//
+//
+// Uncomment when this JEP status is set to Accepted, Rejected or Withdrawn.
+//| Resolution
+//| Link to relevant post in the jenkinsci-dev@ mailing list archives
+
+|===
+
+== Abstract
+
+This proposal makes it possible to run Jenkins in such a way that it would be link:https://csrc.nist.gov/pubs/fips/140-2/upd2/final[FIPS-140] compliant.  Not all features may be possible to use in this mode of operation, and plugins may need to change their behaviour at runtime.
+Configuration of the JVM, platform etc. is out-of-scope, as is making every feature/plugin compliant.
+
+
+== Specification
+
+* A new API will be introduced that code can use to check if FIPS-140 compliance is preferred.
+* Code in Jenkins core (where necessary) switches behaviour between more secure yet non compliant cryptography to FIPS-140 compliant cryptography.
+* Non compliant and less secure crytpography has been replaced by a secure and compliant option.
+* General guidelines exist to help developers write FIPS-140 compliant code and/or understand incompatibilities with the standard.
+* User facing documentation provides a mere overview of the FIPS flag and what it does (and does not do).
+
+== Motivation
+
+Currently it is not possible to run Jenkins in a link:https://csrc.nist.gov/pubs/fips/140-2/upd2/final[FIPS-140-2] compliant way.  Many users needing Jenkins are required (e.g. for US government projects) to have any solution be compliant with FIPS-140 and today it is not possible for them to do this. 
+
+Both Jenkins core and plugins use cryptography that is not allowed and/or are not from an approved implementation, and this prevents adoption of Jenkins by these end users.
+Examples would be during setup where the install wizard creates a first admin user using the internal security realm which uses ``bcrypt`` which would fall fowl of the standard due to its use of ``blowfish`` which is not an approved algorithm.
+
+== Reasoning
+
+In order to be FIPS-140 compliant the encryption algorithms used need to be validated, this requires complex configuration of the host, the JVM as well as the changes to Jenkins and plugin code as well as checking of any plugin code that would be installed.  As such it is not considered that the Jenkins community supports, or contains instructions on how to run Jenkins in a compliant manner, but merely contains only an overview to say it may be possible with various configuration of the JVM, Servlet environment, limitations of plugins and finally instructing Jenkins to prefer a FIPS-140 mode of operation.
```

Comment:
The sentence seems like it contains multiple concepts that could be split into separate sentences.  Google Bard (and some edits of my own) suggested the following sentences.  Optional feedback that may use the wrong words or the wrong concepts.  You're welcome to ignore the comment
```suggestion
To be FIPS-140 compliant, Jenkins must use validated, FIPS-140 compliant encryption algorithms. This requires complex configuration of the host, the Java Virtual Machine (JVM), the servlet environment, and the Jenkins controller. It requires changes to Jenkins core and to Jenkins plugins.. As such it is not considered that the Jenkins community supports or provides instructions on how to run Jenkins in a compliant manner, but merely states that FIPS-140 compliant operation may be  possible with appropriate configuration of the JVM, Servlet environment, limited set of plugins, and finally instructing Jenkins to prefer a FIPS-140 mode of operation.
```